### PR TITLE
fix: instrumented broker is crashing when calling aedes.subscribe

### LINKED ISSUE #2: [bug]: instrumented broker is crashing when calling `aedes.subscribe`
**Describe the bug**

The client argument is undefined when calling aedes.subscribe directly

**Expected behavior**

Allow patched methods to work where clients might be undefined

### PR DESCRIPTION
**Pull request by getlarge**
- fixes #2 
- setup `@release-it/conventional-changelog` to generate changelog on releases

### CODE REVIEW

File: src/aedes-instrumentation.ts [hunk 1]
```diff
@@ -155,31 +167,44 @@ export class AedesInstrumentation extends InstrumentationBase {
 
   // #region span management
 
-  private addClientSpan(client: AedesClient, span: Span) {
+  // TODO: maybe handle case where client is undefined (meaning the span is linked to the broker) ?
```
Comment (robertsLando) [inline src/aedes-instrumentation.ts:170]: I would handle this too, broker has also an Id (more useful in cluster context), we could create a fake client with id `$BROKER_${broker.id}`

**Review by robertsLando — CHANGES_REQUESTED** [review]

**Comment by robertsLando** [comment]
Also I would try do do a dry run with release-it because I dunno if that works when changlog file is not present, maybe you have to firstly generate it as the release-it plugin should work in append mode

File: src/aedes-instrumentation.ts (hunk 1 above)
Comment (getlarge) [reply src/aedes-instrumentation.ts:170]: i get the idea and agree with it, that would be quite a refactoring of the span management logic.
right now, spans are only focused around the client, changing this to also support the broker as the client would involve several changes, beyond changing the functions signatures.
That would also involve :
- checking when the client is the broker 
- storing some attributes into the broker to reuse during span creation
- ensuring that the broker instance is always accessible to manage spans, even though that one should be easy if we consider that the places where “broker is the client” is when calling the broker instance methods
- adding some tests obviously

I would rather do this work in a separate issue/PR.

**Comment by getlarge** [comment]
> Also I would try do do a dry run with release-it because I dunno if that works when changlog file is not present, maybe you have to firstly generate it as the release-it plugin should work in append mode

it looks like this :

## [0.1.1](https://github.com/moscajs/aedes-otel-instrumentation/compare/v0.1.0...v0.1.1) (2023-09-25)


### Bug Fixes

* correctly create and close spans when no client is defined in Aedes methods ([961db1a](https://github.com/moscajs/aedes-otel-instrumentation/commit/961db1ae23676938251e067b8b2dc4b747511a6f))


### Test added

* ensure broker.subscribe can be used by consumer ([15de1a4](https://github.com/moscajs/aedes-otel-instrumentation/commit/15de1a4272734ba7922d8043a40694e03156cc9d))


### Chores

* add changelog generator ([8970b43](https://github.com/moscajs/aedes-otel-instrumentation/commit/8970b4328ae01b21fe54f8176db12673da2f5e09))

File: src/aedes-instrumentation.ts (hunk 1 above)
Comment (robertsLando) [reply src/aedes-instrumentation.ts:170]: ok, agree 👍🏼 Issue so?

**Comment by robertsLando** [comment]
> it looks like this :

Perfect so!

**Review by robertsLando — APPROVED** [review]

File: src/aedes-instrumentation.ts (hunk 1 above)
Comment (getlarge) [reply src/aedes-instrumentation.ts:170]: #5

**Pull request merged**